### PR TITLE
example: add semantic cache demo built with jinaai and tidb vector

### PR DESCRIPTION
**Pull request by sykp241095**

### CODE REVIEW

File: examples/semantic-cache/cache.py
Comment (wd0517) [inline examples/semantic-cache/cache.py:127]: The ANN index will not be enabled when using a pre-filter.

File: examples/semantic-cache/cache.py
Comment (wd0517) [inline examples/semantic-cache/cache.py:11]: Please use `AsyncSession`

File: examples/semantic-cache/cache.py
Comment (wd0517) [inline examples/semantic-cache/cache.py:79]: not used

File: examples/semantic-cache/cache.py
Comment (wd0517) [inline examples/semantic-cache/cache.py:50]: Provide a document link to introduce it.

File: examples/semantic-cache/cache.py
Comment (sykp241095) [reply examples/semantic-cache/cache.py:11]: I removed all `async def` controller, so there is no need to use AsyncSession/AsyncEngine anymore.

File: examples/semantic-cache/cache.py
Comment (sykp241095) [reply examples/semantic-cache/cache.py:126]: @wd0517 Any suggestions about the pre-filter or post-filter?

Comment (wd0517) [reply examples/semantic-cache/cache.py:126]: don't use filter in sql, filter in code

**Comment by sykp241095** [comment]
@wd0517 You should approve this before we can merge it as there is a branch protection policy...

**Review by wd0517 — APPROVED** [review]

**Pull request merged**
